components/plurals/src/provider.rs

```diff
@@ -433,50 +444,561 @@ impl<'a> PluralElementsV1<'a> {
+    fn to_plural_elements(&self) -> PluralElements<(FourBitMetadata, &V)> {
+        let (default, specials) = self.as_parts();
+        let mut elements = PluralElements::new(default);
+        if let Some(specials) = specials {
+            for special in specials.iter() {
+                let (key, metadata) = special.sized.get();
+                let value = (metadata, &special.variable);
+                match key {
+                    PluralElementsKeysV1::Zero => elements.zero = Some(value),
+                    PluralElementsKeysV1::One => elements.one = Some(value),
+                    PluralElementsKeysV1::Two => elements.two = Some(value),
+                    PluralElementsKeysV1::Few => elements.few = Some(value),
+                    PluralElementsKeysV1::Many => elements.many = Some(value),
+                    PluralElementsKeysV1::ExplicitZero => elements.explicit_zero = Some(value),
+                    PluralElementsKeysV1::ExplicitOne => elements.explicit_one = Some(value),
+                }
+            }
+        }
+        elements
+    }
+}
+
+impl<T> PluralElements<(FourBitMetadata, T)>
+where
+    T: PartialEq,
+{
+    fn to_packed_builder<'a, V>(
+        &'a self,
+    ) -> Result<(FourBitMetadata, PluralElementsPackedBuilder<'a, T, V>), TinyVarVarError>
+    where
+        &'a T: EncodeAsVarULE<V>,
+        V: VarULE + ?Sized,
+    {
+        if self.has_specials() {
+            Ok((
+                self.other().0,
+                PluralElementsPackedBuilder::Multi(TinyVarVar::try_new(
+                    &self.other().1,
+                    self.get_specials_tuple_vec()
+                        .map(|(key, (metadata, t))| VarTuple {
+                            sized: PluralCategoryAndMetadata::new(key, *metadata),
+                            variable: t,
+                        })
+                        .collect::<Vec<_>>(),
+                )?),
+            ))
+        } else {
+            Ok((
+                self.other().0,
+                PluralElementsPackedBuilder::Singleton(&self.other().1),
+            ))
+        }
+    }
+}
+
+unsafe impl<T, V> EncodeAsVarULE<PluralElementsPackedULE<V>>
+    for PluralElements<(FourBitMetadata, T)>
+where
+    T: PartialEq + fmt::Debug,
+    for<'a> &'a T: EncodeAsVarULE<V>,
+    V: VarULE + ?Sized,
+{
+    fn encode_var_ule_as_slices<R>(&self, _cb: impl FnOnce(&[&[u8]]) -> R) -> R {
+        // unnecessary if the other two are implemented
+        unreachable!()
+    }
+
+    fn encode_var_ule_len(&self) -> usize {
+        match self.to_packed_builder::<V>() {
+            Ok((_, PluralElementsPackedBuilder::Singleton(v))) => 1 + v.encode_var_ule_len(),
+            Ok((_, PluralElementsPackedBuilder::Multi(v))) => 1 + v.encode_var_ule_len(),
+            Err(_) => {
+                // TODO: Inform the user more nicely that their data doesn't fit in our packed structure
+                panic!("other value too long to be packed: {self:?}")
```

Comment:
One way I could get rid of the explicit panic would be to restrict the PatternElement's T to be a certain length (create a newtype such as ShortT).